vnext/Shared/InspectorPackagerConnection.cpp

```diff
@@ -0,0 +1,233 @@
+    }
+
+    if (0 == event.compare(Message_eventName_connect)) {
+      return EventType::Connect;
+    }
+
+    if (0 == event.compare(Message_eventName_disconnect)) {
+      return EventType::Disconnect;
+    }
+
+    assert(false && "Unknown event!");
+    std::abort();
+  }
+
+  static folly::dynamic constructResponseForPackager(EventType eventType, folly::dynamic &&payload) {
+    folly::dynamic response = folly::dynamic::object;
+
+    switch (eventType) {
+      case EventType::GetPages:
+        response[InspectorProtocol::Message_EVENT] = InspectorProtocol::Message_eventName_getPages;
+        break;
+      case EventType::WrappedEvent:
+        response[InspectorProtocol::Message_EVENT] = InspectorProtocol::Message_eventName_wrappedEvent;
+        break;
+      case EventType::Connect:
+        response[InspectorProtocol::Message_EVENT] = InspectorProtocol::Message_eventName_connect;
+        break;
+      case EventType::Disconnect:
+        response[InspectorProtocol::Message_EVENT] = InspectorProtocol::Message_eventName_disconnect;
+        break;
+      default:
+        assert(false && "Unknown event Type.");
+        std::abort();
+    }
+
+    response[InspectorProtocol::Message_PAYLOAD] = std::move(payload);
+    return response;
+  }
+
+  static folly::dynamic constructGetPagesResponsePayloadForPackager(
+      const std::vector<facebook::react::InspectorPage> &pages) {
+    folly::dynamic payload = folly::dynamic::array;
+    for (const facebook::react::InspectorPage &page : pages) {
+      folly::dynamic pageDyn = folly::dynamic::object;
+      pageDyn["id"] = page.id;
+      pageDyn["title"] = page.title;
+      pageDyn["vm"] = page.vm;
+
+      pageDyn["isLastBundleDownloadSuccess"] = true;
+      pageDyn["bundleUpdateTimestamp"] = "0";
+
+      payload.push_back(pageDyn);
+    }
+    return payload;
+  }
+
+  static folly::dynamic constructVMResponsePayloadForPackager(int64_t pageId, std::string &&messageFromVM) {
+    folly::dynamic payload = folly::dynamic::object;
+    payload[InspectorProtocol::Message_eventName_wrappedEvent] = messageFromVM;
+    payload[InspectorProtocol::Message_PAGEID] = pageId;
+    return payload;
+  }
+
+  static folly::dynamic constructVMResponsePayloadOnDisconnectForPackager(int64_t pageId) {
+    folly::dynamic payload = folly::dynamic::object;
+    payload[InspectorProtocol::Message_PAGEID] = pageId;
+    return payload;
+  }
+};
+
+} // namespace
+
+RemoteConnection::RemoteConnection(int64_t pageId, const InspectorPackagerConnection &packagerConnection)
+    : m_packagerConnection(packagerConnection), m_pageId(pageId) {}
+
+void RemoteConnection::onMessage(std::string message) {
+  folly::dynamic response = InspectorProtocol::constructResponseForPackager(
+      InspectorProtocol::EventType::WrappedEvent,
+      InspectorProtocol::constructVMResponsePayloadForPackager(m_pageId, std::move(message)));
+  std::string responsestr = folly::toJson(response);
```

Comment:
Just for cosnideration:
I realize this is folly's design. 
I'm not sure if this protocol is super chatty, perhaps a json writer style serialization would justify some extra complexity for perf.

Reply:
Inspector internally use the same folly helpers to construct the JSON messages .. but the inspector API that we interact with (RemoteConnection::onMessage for responses) doesn't have folly constructs in it .. Hence, we can use some other libraries. One option is the JSON reader/writer library that @vmoroz has written for Reka. We could consider it if there are perf concerns. The CDP protocol can be very chatty depending on the usage.